assets/app/scripts/directives/oscResourceName.js

```diff
@@ -0,0 +1,35 @@
+"use strict";
+
+angular.module("openshiftConsole")
+  .directive("oscResourceName", function(){
```

Comment:
is this directive only adding a validator?  if so might be nice to call it oscResourceNameValidator, then it is clear in the templates where you are using it that is what it is adding